### PR TITLE
DDF-5056 added hidden and conditional search forms to configuration application

### PR DESCRIPTION
**Pull request by Corey-Collins**
#### What does this PR do?
Adds two configs to the `/config` endpoint.
`hiddenSearchForms` to list which forms are hidden in the ui
`conditionalSearchForms` to list which forms are hidden in the ui based on the user's preferences
#### Who is reviewing it? 

@andrewzimmer 
@zta6 
@hayleynorton 
@willwill96 
#### Select relevant component teams: 
@codice/core-apis 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining
@millerw8

#### What are the relevant tickets?
Fixes: #5056 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/resources/OSGI-INF/metatype/org.codice.ddf.catalog.ui.xml [hunk 1]
```diff
@@ -322,6 +322,21 @@
             cardinality="100"
             required="false">
         </AD>
+
+        <AD id="hiddenSearchForms"
+            name="Hidden Search Forms"
+            description="List of hidden search forms"
+            type="String"
+            cardinality="100"
```
Comment (Corey-Collins) [inline catalog/ui/catalog-ui-search/src/main/resources/OSGI-INF/metatype/org.codice.ddf.catalog.ui.xml]: remove/adjust cardinalities

**Comment by rzwiefel** [comment]
Are these all based on user preferences Or is this some global that an admin can set to hide certain search forms? I'd imagine the user preferences for search forms would go into the user prefs object instead of the global config endpoint - unless i'm misunderstanding what this is doing

**Comment by andrewzimmer** [comment]
@rzwiefel this change is to allow certain search forms to be hidden based on the current user environment

**Comment by rymach** [comment]
dumb question - where is the code that actually uses these configs?

**Comment by andrewzimmer** [comment]
@rymach I have a pr in progress that will leverage these configs

**Pull request closed without merging**